core/src/main/resources/META-INF/resources/primefaces-extensions/session/session.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * PrimeFaces Extensions Session Widget.
+ *
+ * @author Frank Cornelis
+ * @since 12.0
+ */
+
+PrimeFaces.widget.Session = PrimeFaces.widget.BaseWidget.extend({
+
+    init: function (cfg) {
+        this._super(cfg);
+        this.configureTimer();
+        this.registerAjaxCallbacks();
+    },
+
+    registerAjaxCallbacks() {
+        let _this = this;
+        jsf.ajax.addOnEvent(() => {
+            _this.jsfAjaxEventCallback();
+        });
+        // PrimeFaces does its own client-side lifecycle.
+        let doc = $(document);
+        doc.on('pfAjaxComplete', () => {
+            _this.jsfAjaxEventCallback();
+        });
+    },
+
+    refresh: function (cfg) {
+        this.cancelTimers();
+        this._super(cfg);
+    },
+
+    configureTimer() {
+        this.cancelTimers();
+        if (typeof this.cfg.max_inactive_interval !== "undefined") {
+            let timeout = (Number(this.cfg.max_inactive_interval) - this.cfg.reactionPeriod) * 1000;
+            if (timeout > 0) {
+                let _this = this;
+                console.log("setting timeout for: " + timeout + " ms.");
+                this.timer = window.setTimeout(() => {
+                    _this.timeoutCallback();
+                }, timeout);
+            }
+        }
+    },
+
+    timeoutCallback() {
+        this.timer = null;
+        eval(this.cfg.onexpire);
```

Comment:
Don't use `eval` it does't play nicely with CSP once you fix the above call it like Clipboard does ...

```js
            if (this.cfg.onexpire) {
                this.cfg.onexpire.call();
            }
```